Pixel Game/Assets/Scripts/Damage.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Damage : MonoBehaviour
+{
+    public enum Group { Players, Mobs }
+
+    [SerializeField]
+    private int attackPower = 0;
+    [SerializeField]
+    private float attackCooldown = 1;
+    [SerializeField]
+    private Group group;
+    [SerializeField]
+    private GameObject damageReceiver;
+    
+    private DateTime lastAttackTime;
+    private Health health;
+
+    private bool CanInflictDamage => attackPower > 0;
+    private bool CanTakeDamage => damageReceiver != null;
+
+    private void Start()
+    {
+        if (damageReceiver != null)
+            health = damageReceiver.GetComponent<Health>();
+    }
+    //Inflict damage on colliding object
+    private void OnTriggerStay2D(Collider2D collider)
```

Comment:
Is there a reason why we use OnTrigger**Stay**2D instead of OnTrigger**Enter**2D??

Reply:
It's because next attack should start as soon as the attack cooldown is finished, without having to exit and re-enter collision.

Reply:
Ok cool!